debian/apport-general-hook.py

```diff
@@ -0,0 +1,51 @@
+import json
+import logging
+
+
+def get_ec2_data(instance_ci_data):
+    """
+    provided cloud-init data from /run/cloud/instance-data.json
+    populate a smaller set of information helpful for debugging
+    """
+    stripped_data = {}
+    instance_identity_keys = (
+        "architecture",
+        "billingProducts",
+        "imageId",
+        "region",
+    )
+    metadata_keys = ("instance-type",)
+    for key in instance_identity_keys:
+        stripped_data[key] = instance_ci_data["ds"]["dynamic"][
+            "instance-identity"
+        ]["document"][key]
+    for key in metadata_keys:
```

Comment:
What's the motive for iterating over a tuple with length == 1?

Reply:
Probably to easily extent it in the future. I changed that.